File: Frontend/src/server-models/cqrs/authorization/responses/sign-in.response.ts
import { Subscription } from 'src/server-models/stripe/subscription.model';
import { AccountStatus } from '../../../enums/account-status.enum';
import { SubscriptionStatus } from '../../../enums/subscription-status.enum';

export class SignInResponse {
     id: string;
     customerId: string;
     username: string;
     firstName: string;
     lastName: string;
     fullName: string;
     email: string;
     accountType: string;
     accountStatus: AccountStatus;
     subscriptionStatus: SubscriptionStatus;
     subscriptionInfo: Subscription;
     trialDaysRemaining: number;
}

File: Backend/Core/Backend.Application.Business/Business/Authorization/Commands/SignIn/SignInResponse.cs
using System;
using Backend.Domain.Enum;
using Backend.Service.Payment.Enums;
using Stripe;
using AccountType = Backend.Domain.Enum.AccountType;

namespace Backend.Application.Business.Business.Authorization.Commands.SignIn
{
    public class SignInResponse
    {
        public SignInResponse()
        {

        }

        public Guid Id { get; set; }
        public string CustomerId { get; set; }
        public string Username { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string FullName => $"{FirstName} {LastName}";
        public string Email { get; set; }
        public AccountType AccountType { get; set; }
        public AccountStatus AccountStatus { get; set; }
        public SubscriptionStatus SubscriptionStatus { get; set; }
        public Subscription SubscriptionInfo { get; set; }
        public int TrialDaysRemaining { get; set; }
    }
}
